README
NAME
    Perl::Dist - Perl Distribution Creation Toolkit

DESCRIPTION
    The Perl::Dist namespace encompasses creation of pre-packaged, binary
    distributions of Perl, primarily as executable installers for Win32.

    Packages in this namespace include both "builders" and "distributions".

    Builder packages automate the generation of distributions.

    Distribution packages contain configuration files for a particular
    builder, extra files to be bundled with the pre-packaged binary, and
    documentation.

    Distribution namespaces are also recommended to consolidate bug
    reporting using http://rt.cpan.org/.

    *Distribution packages should not contain the pre-packaged install files
    themselves.*

  BUILDERS
    At the present time the primarily builder module is Perl::Dist::Inno.

  DISTRIBUTIONS
    Currently available distributions include:

    *   Perl::Dist::Vanilla -- An experimental "core Perl" distribution
        intended for distribution developers.

    *   Perl::Dist::Strawberry -- A practical Win32 Perl release for
        experienced Perl developers familiar with Perl on Unix environments
        with full CPAN capabilities.

        Strawberry Perl is considered stable, and can be downloaded from the
        Strawberry Perl website at <http://strawberryperl.com/>.

    *   Perl::Dist::Chocolate -- A concept distribution that bundled a large
        "standard library" collection of CPAN modules, and provides a
        variety of WxWindows-based GUI tools.

    *   Perl::Dist::Bootstrap -- Bootstrap Perl is a Perl 5.8.8 distribution
        designed for people that are themselves creating Perl distributions.

        It installs to a disk location out of the way and not used by any
        "end-user" distributions, and comes with Perl::Dist and support
        modules pre-bundled.

ROADMAP
    Perl::Dist::Inno, based on Inno Setup, is working well, but has a
    limited lifespace, as it is not capable of produces Windows native .msi
    installer files.

    Ultimately this means that Perl::Dist will see the additional of an
    alternative module based on Nullsoft or Wix that allows the creation of
    .msi files (a major feature for corporate users).

    Various other features are able to be implemented within the Inno Setup
    feature set, and include:

    *   Bug-squashing Win32 compatibility problems in popular modules

    *   Customisable installation path.

    *   Support installation paths with spaces and other weird characters.

    *   Restore support for .exe installation instead of .zip.

    *   Better uninstall support and upgradability.

SUPPORT
    Bugs should be reported via the CPAN bug tracker at

    <http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Perl-Dist>

AUTHORS
    Adam Kennedy <adamk@cpan.org>

    David A. Golden <dagolden@cpan.org>

COPYRIGHT
    Copyright 2007 - 2008 Adam Kennedy.

    Copyright 2006 David A. Golden.

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.

    The full text of the license can be found in the LICENSE file included
    with this module.

SEE ALSO
    Perl::Dist, Perl::Dist::Vanilla, Perl::Dist::Strawberry,
    <http://win32.perl.org/>, <http://vanillaperl.com/>,
    <irc://irc.perl.org/#win32>, <http://ali.as/>, <http://dagolden.com/>



Changes
Revision history for Perl module Perl-Dist

1.12 Thu 29 Jan 2009
	- This release co-incides with Strawberry Perl January 2009
	- Various minor tweaks

1.11 Thu  4 Dec 2008
	- This release represents a moment of stability at 5.10.0.4 Beta 1
	- Adding an improved caching useragent
	- Some minor tweaks to allow Parrot builds

1.10 Tue 11 Nov 2008
	- Adding initial support for Perl 5.8.9

1.09 Mon 20 Oct 2008
	- This version was used to build Strawberry Perl October 2008

1.08 Sat 18 Oct 2008
        - Shortcut run when no zip or exe target

1.07 Thu 16 Oct 2008
	- Better linking to Perl Documentation

1.06 Wed 15 Oct 2008
	- This is expected to be the Perl::Dist used to build
	  Strawberry Perl October
	- Bootstrap Perl is moving to the Strawberry distribution
	- After the toolchain is installed, run a full CPAN upgrade
	  to catch anything out of date NOT explicitly listed in the
	  toolchain list.
	- All tests (finally) pass during the 5 hours test run

1.05_04 Tue 14 Oct 2008
	- Adding missing dependency on File::HomeDir
	- Additional minor tweaks and minor bug squashing

1.05_03 Mon 13 Oct 2008
	- Additional tweaks and clean ups
	- Force the install of CPAN.pm

1.05_02 Sat 11 Oct 2008
	- Very aggressive dependencies, everything now to maximum
	- Discovered and fixed a number of dependency bugs
	- Bug fix in 09_file_asset.t

1.05_01 Sat 11 Oct 2008
	- Reorganising the test script order
	- Added separate 5.8.8 and 5.10.0 test scripts
	- Added a full test of Vanilla ->run
	- Removing force flags for stuff that now works
	- Adding a new Template-Toolkit-based patch system
	- Unified the toolchain install methods
	- Set the environment correctly when installing CPAN modules

1.04 Wed 23 Jul 2008
	- This release co-incides with Strawberry August 2008
	- Tweak to allow libwww-perl to build behind a proxy
	- Add a temporary bug-fix/patch for MM_Win32.pm
	- Add a workaround for Data::Compare failing on Win32
	- Adding Math::Pari to the default build

1.03 Tue 22 Jul 2008
	- Minor cleanups to support the July Strawberry release

1.02 Tue 15 Jul 2008
	- Updating to a Vista-capable mingw.
	- Adding install_pari for Math::Pari support

1.01 Tue 15 Apr 2008
	- Refactored a bit to use install_modules
	- Promoting some internals to public methods

1.00 Mon  7 Apr 2008
	- Release to match Strawberry Perl 5.10.0.1 April 2008

0.90_04 Mon 24 Mar 2008
	- Adding Perl::Dist::Inno::System for [Run] and [UninstallRun]
	  block support.
	  This is needed for more sophisticated install and uninstall
	  functionality.
	- Completing documentation for all the major install_foo methods.
	- Merging some idiosyncracies into the Perl::Dist::Inno core.
	- Converting force in install_module to use notest under the covers,
	  so that there are no errors to propogate upwards.
	- Added support for generating both zip and exe files simultaneously.

0.90_03 Fri  4 Jan 2008
	- Perl::Dist::Download was a bad idea, I wanted the ability to
	  to build distribution from a minicpan mirror, but maintaining
	  that package and incrementing it every time a minor release
	  happens of an upstream package will become a nightmare.
	- Merged in PAR support from Perl-Dist-Glutamate (Steffen Muller)
	- Adding more documentation
	- Completed first implementation of Perl::Dist::Machine, for
	  generating multiple variations of a dist at the same time.
	- This version used to create Vanilla Perl Build 15.
	- This version implements features needed for Strawberry Perl April.

0.90_02 Thu  3 Jan 2008
	- This release was used to build Vanilla Perl 14
	- Reorganising the default way that app_ver_name etc are done
	  to allow for distributions classes that can generate BOTH
	  5.8.8 and 5.10.0 distributions.
	- Changed the usage of perldist
	- Make sure Perl::Dist::Util::Toolchain in the tests uses the
	  full explicit Perl path.

0.90_01 Wed  2 Jan 2008
	- Major API refactoring
	- Converted Perl::Dist::Util::Toolchain to be completely
	  auto-resolving for what it should install.
	- Started the documentation process.

0.53 Sun 24 Dec 2007 (unreleased)
	- Adding support for websites in the start menu entries
	- Adding support for .bat launchers in the start menu
	- Completed Strawberry Perl 5.10.0 using this version
	- Wide-ranging refactoring and documentation to make
	  Perl::Dist suitable for other people to start using
	  for making Perl distributions.

0.52 Thu 20 Dec 2007
        - Adding an additional install flag to prevent
          installation to the old directory.

0.51 Wed 19 Dec 2007
	- Hard-code in the path to Perl 5.10.0 final
	- Move default environment vars from ::Script to ::Inno
	- Add share to no_index to prevent 5.8.8 overwritten
	  modules being listed on search.cpan.org

0.50 Wed 19 Dec 2007
	- Merging in Perl::Dist::Vanilla
	- Merging in Perl::Dist::Bootstrap
	- Added Perl::Dist::Util::Toolchain
	- Dynamically generate the distribution list
	- Clean up the shared files
	- Bug fixing and repairing perldist
	- Completed generation of exe releases

0.30 Sun 18 Nov 2007
	- Adding support for Perl 5.10.0
	- Adding support for zlib
	- Adding support for bzlib2
	- Adding support for iconv
	- Adding support for libxml2
	- Adding support for SQLite
	- Completing generation of zip releases

0.29_03 Fri  9 Nov 2007
	- Completing install_perl_588
	- Completing install_distribution
	- Completing install_module
	- Adding install_library
	- Adding install_libraries
	- Completing installation of zlib
	- Starting installation of iconv
	- Adding pexports to the list of binaries to install

0.29_02 Thu 31 Oct 2007
        - Adding install_file support for tweaking stuff
          post-installation of binaries/perl/dists/etc
        - Removed post_copy and replaced with install_file
        - Renamed pre_copy to patch since it makes more sense
          that way now.

0.29_01 Wed 31 Oct 2007
	- Experimental release, only intended for review
	- Huge refactoring
	
0.10  Fri 19 Oct 2007
        - Converting from config-driver to code-driven

t/lib/Test.pm
package t::lib::Test;

# Generic base class for test classes

use strict;
use File::Spec::Functions ':ALL';
use Test::More    ();
use File::Path    ();
use File::Remove  ();
use t::lib::Test1 ();
use t::lib::Test2 ();
use t::lib::Test3 ();
use t::lib::Test4 ();
use t::lib::Test5 ();

use vars qw{$VERSION};
BEGIN {
	$VERSION = '1.12';
}





#####################################################################
# Default Paths

sub make_path {
	my $dir = rel2abs( catdir( curdir(), @_ ) );
	File::Path::mkpath( $dir ) unless -d $dir;
	Test::More::ok( -d $dir, 'Created ' . $dir );
	return $dir;
}

sub remake_path {
	my $dir = rel2abs( catdir( curdir(), @_ ) );
	File::Remove::remove( \1, $dir ) if -d $dir;
	File::Path::mkpath( $dir );
	Test::More::ok( -d $dir, 'Created ' . $dir );
	return $dir;
}

sub paths {
	my $class        = shift;
	my $subpath      = shift || '';
	my $basedir      = rel2abs( catdir( 't', "tmp$subpath" ) );
	# File::Remove::clear( $basedir );
	my $output_dir   = remake_path( catdir( $basedir, 'output'   ) );
	my $image_dir    = remake_path( catdir( $basedir, 'image'    ) );
	my $download_dir =   make_path( catdir( $basedir, 'download' ) );
	my $build_dir    = remake_path( catdir( $basedir, 'build'    ) );
	return (
		output_dir   => $output_dir,
		image_dir    => $image_dir,
		download_dir => $download_dir,
		build_dir    => $build_dir,
	);
}

sub cpan {
	if ( $ENV{TEST_PERLDIST_CPAN} ) {
		return URI->new($ENV{TEST_PERLDIST_CPAN});
	}
	my $path = rel2abs( catdir( 't', 'data', 'cpan' ) );
	Test::More::ok( -d $path, 'Found CPAN directory' );
	Test::More::ok( -d catdir( $path, 'authors', 'id' ), 'Found id subdirectory' );
	return URI::file->new($path . '\\');
}

sub new1 {
	my $class = shift;
	return t::lib::Test1->new(
		cpan => $class->cpan,
		$class->paths(@_),
	);
}

sub new2 {
	return t::lib::Test2->new( shift->paths(@_) );
}

sub new3 {
	return t::lib::Test3->new( shift->paths(@_) );
}

sub new4 {
	return t::lib::Test4->new( shift->paths(@_) );
}

sub new5 {
	return t::lib::Test5->new( shift->paths(@_) );
}

1;


t/lib/Test1.pm
package t::lib::Test1;

use strict;
use Perl::Dist::Inno;

use vars qw{$VERSION @ISA};
BEGIN {
	$VERSION = '1.12';
	@ISA     = 'Perl::Dist::Inno';
}





#####################################################################
# Configuration

sub app_name             { 'Test Perl'               }
sub app_ver_name         { 'Test Perl 1 alpha 1'     }
sub app_publisher        { 'Vanilla Perl Project'    }
sub app_publisher_url    { 'http://vanillaperl.org'  }
sub app_id               { 'testperl'                }
sub output_base_filename { 'test-perl-5.8.8-alpha-1' }





#####################################################################
# Main Methods

sub new {
	return shift->SUPER::new(
		perl_version => 588,
		@_,
	);
}

sub run {
	my $self = shift;

	# Just install a single binary
	$self->checkpoint_task( install_dmake => 1 );

	return 1;
}

sub trace { 1 } # Test::More::diag($_[1]) }

sub install_binary {
	return shift->SUPER::install_binary( @_, trace => sub { 1 } );
}

1;


t/lib/Test2.pm
package t::lib::Test2;

use strict;
use Perl::Dist ();

use vars qw{$VERSION @ISA};
BEGIN {
	$VERSION = '1.12';
	@ISA     = 'Perl::Dist';
}





#####################################################################
# Configuration

sub app_name             { 'Test Perl'               }
sub app_ver_name         { 'Test Perl 1 alpha 1'     }
sub app_publisher        { 'Vanilla Perl Project'    }
sub app_publisher_url    { 'http://vanillaperl.org'  }
sub app_id               { 'testperl'                }
sub output_base_filename { 'test-perl-5.8.8-alpha-1' }





#####################################################################
# Main Methods

sub new {
	return shift->SUPER::new(
		perl_version => 588,
		@_,
	);
}

sub run {
	my $self = shift;

	# Install the core binaries
	$self->install_c_toolchain;

	# Install the extra libraries
	$self->install_c_libraries;

	# Install Perl 5.8.8
	$self->install_perl_588;

	# Install a test distro
	$self->install_distribution(
		name => 'ADAMK/Config-Tiny-2.12.tar.gz',
	);

	return 1;
}

sub trace { Test::More::diag($_[1]) }

sub install_binary {
	return shift->SUPER::install_binary( @_, trace => sub { 1 } );
}

sub install_distribution {
	return shift->SUPER::install_distribution( @_, trace => sub { 1 } );
}

sub install_file {
	return shift->SUPER::install_file( @_, trace => sub { 1 } );
}

sub install_perl_588_bin {
	return shift->SUPER::install_perl_588_bin( @_, trace => sub { 1 } );
}

sub install_perl_588_toolchain {
	return shift->SUPER::install_perl_588_toolchain( @_, trace => sub { 1 } );
}

1;


lib/Perl/Dist.pm
package Perl::Dist;

use 5.006;
use strict;
use Perl::Dist::Inno ();

use vars qw{$VERSION @ISA};
BEGIN {
        $VERSION = '1.12';
	@ISA     = 'Perl::Dist::Inno';
}

1;

=pod

=head1 NAME

Perl::Dist - Perl Distribution Creation Toolkit

=head1 DESCRIPTION

The Perl::Dist namespace encompasses creation of pre-packaged, binary
distributions of Perl, primarily as executable installers for Win32.

Packages in this namespace include both "builders" and "distributions".

Builder packages automate the generation of distributions.

Distribution packages contain configuration files for a particular builder,
extra files to be bundled with the pre-packaged binary, and documentation.

Distribution namespaces are also recommended to consolidate bug reporting
using http://rt.cpan.org/.

I<Distribution packages should not contain the pre-packaged install files
themselves.>

=head2 BUILDERS

At the present time the primarily builder module is L<Perl::Dist::Inno>.

=head2 DISTRIBUTIONS

Currently available distributions include:

=over

=item *

L<Perl::Dist::Vanilla> -- An experimental "core Perl" distribution intended
for distribution developers.

=item *

L<Perl::Dist::Strawberry> -- A practical Win32 Perl release for
experienced Perl developers familiar with Perl on Unix environments
with full CPAN capabilities.

Strawberry Perl is considered stable, and can be downloaded from the
Strawberry Perl website at L<http://strawberryperl.com/>.

=item *

L<Perl::Dist::Chocolate> -- A concept distribution that bundled a large
"standard library" collection of CPAN modules, and provides a variety of
WxWindows-based GUI tools.

=item *

L<Perl::Dist::Bootstrap> -- Bootstrap Perl is a Perl 5.8.8 distribution
designed for people that are themselves creating Perl distributions.

It installs to a disk location out of the way and not used by any "end-user"
distributions, and comes with Perl::Dist and support modules pre-bundled.

=back

=head1 ROADMAP

L<Perl::Dist::Inno>, based on Inno Setup, is working well, but has a
limited lifespace, as it is not capable of produces Windows native
.msi installer files.

Ultimately this means that Perl::Dist will see the additional of an
alternative module based on Nullsoft or Wix that allows the creation
of .msi files (a major feature for corporate users).

Various other features are able to be implemented within the Inno Setup
feature set, and include:

=over

=item *

Bug-squashing Win32 compatibility problems in popular modules

=item *

Customisable installation path.

=item *

Support installation paths with spaces and other weird characters.

=item *

Restore support for .exe installation instead of .zip.

=item *

Better uninstall support and upgradability.

=back

=head1 SUPPORT

Bugs should be reported via the CPAN bug tracker at

L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Perl-Dist>

=head1 AUTHORS

Adam Kennedy <adamk@cpan.org>

David A. Golden <dagolden@cpan.org>

=head1 COPYRIGHT

Copyright 2007 - 2009 Adam Kennedy.

Copyright 2006 David A. Golden.

This program is free software; you can redistribute
it and/or modify it under the same terms as Perl itself.

The full text of the license can be found in the
LICENSE file included with this module.

=head1 SEE ALSO

L<Perl::Dist>, L<Perl::Dist::Vanilla>,
L<Perl::Dist::Strawberry>, L<http://win32.perl.org/>,
L<http://vanillaperl.com/>, L<irc://irc.perl.org/#win32>,
L<http://ali.as/>, L<http://dagolden.com/>

=cut


inc/Module/Install/Share.pm
#line 1
package Module::Install::Share;

use strict;
use Module::Install::Base;

use vars qw{$VERSION $ISCORE @ISA};
BEGIN {
	$VERSION = '0.76';
	$ISCORE  = 1;
	@ISA     = qw{Module::Install::Base};
}

sub install_share {
	my $self = shift;
	my $dir  = @_ ? pop   : 'share';
	my $type = @_ ? shift : 'dist';
	unless ( defined $type and $type eq 'module' or $type eq 'dist' ) {
		die "Illegal or invalid share dir type '$type'";
	}
	unless ( defined $dir and -d $dir ) {
		die "Illegal or missing directory install_share param";
	}

	# Split by type
	my $S = ($^O eq 'MSWin32') ? "\\" : "\/";
	if ( $type eq 'dist' ) {
		die "Too many parameters to install_share" if @_;

		# Set up the install
		$self->postamble(<<"END_MAKEFILE");
config ::
\t\$(NOECHO) \$(MOD_INSTALL) \\
\t\t"$dir" \$(INST_LIB)${S}auto${S}share${S}dist${S}\$(DISTNAME)

END_MAKEFILE
	} else {
		my $module = Module::Install::_CLASS($_[0]);
		unless ( defined $module ) {
			die "Missing or invalid module name '$_[0]'";
		}
		$module =~ s/::/-/g;

		# Set up the install
		$self->postamble(<<"END_MAKEFILE");
config ::
\t\$(NOECHO) \$(MOD_INSTALL) \\
\t\t"$dir" \$(INST_LIB)${S}auto${S}share${S}module${S}$module

END_MAKEFILE
	}

	# The above appears to behave incorrectly when used with old versions
	# of ExtUtils::Install (known-bad on RHEL 3, with 5.8.0)
	# So when we need to install a share directory, make sure we add a
	# dependency on a moderately new version of ExtUtils::MakeMaker.
	$self->build_requires( 'ExtUtils::MakeMaker' => '6.11' );

	# 99% of the time we don't want to index a shared dir
	$self->no_index( directory => $dir );
}

1;

__END__

#line 125


share/vanilla/CPAN_Config.pm
# This CPAN/Config.pm file has hand-crafted for Vanilla Perl style
# distributions as a good generic product-agnostic default.
# However, the first time you run the CPAN "firsttime" or "o conf init"
# type process, it will overwrite this file and everything will revert
# to flattened and hardcoded strings.




# Find the main distribution paths
use Config ();
my $root     =  $Config::Config{'prefix'};
   $root     =~ s/\\perl$//;
my $cpan     =  "$root\\cpan";
my $minicpan =  "$root\\minicpan";
my $windows  =  $ENV{'SYSTEMROOT'} || "C:\\WINDOWS";
my $system32 =  "$windows\\system32";




# Derive the CPAN Config
$CPAN::Config = {
  applypatch                    => q[],
  auto_commit                   => q[1],
  build_cache                   => q[20],
  build_dir                     => "$cpan\\build",
  build_dir_reuse               => q[0],
  build_requires_install_policy => q[yes],
  bzip2                         => q[],
  cache_metadata                => q[0],
  check_sigs                    => q[0],
  colorize_output               => q[0],
  colorize_print                => q[bold blue on_white],
  colorize_warn                 => q[bold red on_white],
  commandnumber_in_prompt       => q[0],
  cpan_home                     => $cpan,
  curl                          => q[],
  ftp                           => "$system32\\ftp.exe",
  ftp_passive                   => q[1],
  ftp_proxy                     => q[],
  getcwd                        => q[cwd],
  gpg                           => q[],
  gzip                          => q[],
  histfile                      => "$cpan\\histfile",
  histsize                      => q[100],
  http_proxy                    => q[],
  inactivity_timeout            => q[0],
  index_expire                  => q[1],
  inhibit_startup_message       => q[0],
  keep_source_where             => "$cpan\\sources",
  load_module_verbosity         => q[none],
  lynx                          => q[],
  make                          => "$root\\c\\bin\\dmake.exe",
  make_arg                      => q[],
  make_install_arg              => q[UNINST=1],
  makepl_arg                    => "LIBS=-L$root\\c\\lib INC=-I$root\\c\\include",
  mbuild_arg                    => q[],
  mbuild_install_arg            => q[--uninst 1],
  mbuildpl_arg                  => q[],
  ncftp                         => q[],
  ncftpget                      => q[],
  no_proxy                      => q[],
  pager                         => "$system32\\more.com",
  patch                         => q[],
  prefer_installer              => q[MB],
  prefs_dir                     => "$cpan\\prefs",
  prerequisites_policy          => q[follow],
  scan_cache                    => q[atstart],
  shell                         => "$system32\\cmd.exe",
  show_unparsable_versions      => q[0],
  show_upload_date              => q[1],
  show_zero_versions            => q[0],
  tar                           => q[],
  tar_verbosity                 => q[none],
  term_is_latin                 => q[0],
  term_ornaments                => q[0],
  test_report                   => q[0],
  unzip                         => q[],
  urllist                       => [
      q[http://mirrors.kernel.org/CPAN/],
      -d $minicpan ? ($minicpan) : (),
  ],
  use_sqlite                    => q[1],
  wget                          => q[],
  yaml_load_code                => q[0],
  yaml_module                   => q[YAML],
};

1;


Makefile.PL
use inc::Module::Install 0.77;

all_from       'lib/Perl/Dist.pm';
requires       'Archive::Tar'               => '1.42';
requires       'Archive::Zip'               => '1.26';
requires       'File::Spec'                 => '3.29';
requires       'File::Copy::Recursive'      => '0.38';
requires       'File::Find::Rule'           => '0.30';
requires       'File::Path'                 => '2.07';
requires       'File::pushd'                => '1.00';
requires       'File::Remove'               => '1.42';
requires       'File::HomeDir'              => '0.82';
requires       'File::ShareDir'             => '1.00';
requires       'File::PathList'             => '1.04';
requires       'File::Temp'                 => '0.21';
requires       'HTTP::Status'               => '5.817';
requires       'IPC::Run3'                  => '0.042';
requires       'LWP::UserAgent'             => '5.823';
requires       'LWP::UserAgent::WithCache'  => '0.06';
requires       'LWP::Online'                => '1.07';
requires       'Object::Tiny'               => '1.06';
requires       'YAML::Tiny'                 => '1.36';
requires       'Module::CoreList'           => '2.17';
requires       'Params::Util'               => '0.35';
requires       'Template'                   => '2.20';
requires       'CPAN'                       => '1.9301';
requires       'PAR::Dist'                  => '0.42';
requires       'Process'                    => '0.22';
requires       'Process::Storable'          => '0.22';
requires       'Process::Delegatable'       => '0.22';
requires       'IO::Capture'                => '0.05';
requires       'Win32::File::Object'        => '0.01' if $^O eq 'MSWin32';
requires       'Portable::Dist'             => '0.02';
requires       'Probe::Perl'                => '0.01';
test_requires  'Test::More'                 => '0.86';
test_requires  'Test::Script'               => '1.03';
test_requires  'Test::LongString'           => '0.11';
install_script 'perldist';
install_share;

WriteAll;

# Look for Inno Setup
use File::Spec;
print "Looking for Inno Setup 5... ";
unless ( $ENV{PROGRAMFILES} and -d $ENV{PROGRAMFILES} ) {
	pexit("Failed to find the Program Files directory\n");
}
my $innosetup_dir  = File::Spec->catdir( $ENV{PROGRAMFILES}, "Inno Setup 5" );
my $innosetup_file = File::Spec->catfile( $innosetup_dir, 'Compil32.exe' );
unless ( -d $innosetup_dir and -f $innosetup_file ) {
	pexit(
		"failed\n",
		"\n",
		"  Failed to find an installation of Inno Setup 5\n",
		"  Please install http://www.jrsoftware.org/download.php/is.exe\n",
	);
}
print "OK\n";

sub pexit {
	print(@_, "\n");
	exit(0);
}
